distributed/cli/tests/test_dask_worker.py

```diff
@@ -682,3 +684,85 @@ def dask_setup(worker):
         await c.wait_for_workers(1)
         [foo] = (await c.run(lambda dask_worker: dask_worker.foo)).values()
         assert foo == "setup"
+
+
+@pytest.mark.slow
+@pytest.mark.parametrize("nanny", ["--nanny", "--no-nanny"])
+def test_timeout(nanny):
+    worker = subprocess.run(
+        ["dask-worker", "192.168.1.100:7777", nanny, "--death-timeout=1"],
+        text=True,
+        encoding="utf8",
+        capture_output=True,
+    )
+
+    assert "timed out starting worker" in worker.stderr.lower()
+    assert "end worker" in worker.stderr.lower()
+    assert worker.returncode == 1
+
+
+@pytest.mark.slow
+@pytest.mark.skipif(WINDOWS, reason="POSIX only")
+@pytest.mark.parametrize("nanny", ["--nanny", "--no-nanny"])
+@pytest.mark.parametrize("sig", [signal.SIGINT, signal.SIGTERM])
+@gen_cluster(client=True, nthreads=[])
+async def test_signal_handling(c, s, nanny, sig):
```

Comment:
It looks like this test is flaky, my first guess is that the `2 sec` timeout on closing the nannies might lead to unreliable behavior on the CI VMs. Any suggestions on how to avoid this?

Reply:
Is this still flakey?

Reply:
Ran it locally a couple of times without an issue, retriggered CI to double-check.

Reply:
Flakiness should be resolved, no issues on local and two CI runs.